freshmaker/errata.py

```diff
@@ -495,6 +495,28 @@ def get_cve_affected_rpm_nvrs(self, errata_id):
 
         return list(nvrs)
 
+    def get_cve_affected_build_nvrs(self, errata_id, only_modules=False):
+        """ Get module build NVRs which are affected by the CVEs in errata
+
+        :param errata_id: Errata advisory ID, e.g. 25713.
+        :type errata_id: str or int
+        :param only_modules: only return module builds NVR if True.
+        :type only_modules: bool
+        :return: a list of strings each of them is a build NVR
+        :rtype: list
+        """
+        data = self._errata_rest_get(f"/erratum/{errata_id}/builds_by_cve")
+        nvrs = set()
+        for data_by_product in data.values():
+            for product_data in data_by_product.values():
+                for build in product_data.get("builds", []):
+                    for build_info in build.values():
+                        if only_modules and not build_info.get("is_module"):
```

Comment:
Let me return to this. Is this even needed? Isn't it better to check all attached builds? Why would a module advisory have attached anything else than module builds (and vice versa for non-module advisories)?

Reply:
The reason is this is a generic API, others may use this API to get build NVRs from other type of advisories, another choice is renaming this to something like `get_cve_affected_module_nvrs` and the caller should make sure they only call this API against module advisories.
But I'd prefer the current generic API implementation, and there is only `is_module` in Errata's API result, no `is_rpm`, `is_docker` exist, that's why we only need to support `only_modules`

Reply:
My question is: Why not always check all attached builds? I don't understand why some builds should be ignored in a module advisory.

Reply:
I see your concern now, I don't know whether there could be other content other than modules exist in a module advisory, so this is the safest way we can get all modules from a module advisory.